website/src/firebase-utils/login.jsx

```diff
@@ -0,0 +1,63 @@
+import * as firebase from 'firebase';
+
+/** 
+ * Opens login prompt for user and redirects them to the home page if successful.
+ * @param none
+ * @returns none
+ */
+function promptForLogin() {
+  const provider = new firebase.auth.GoogleAuthProvider();
+  firebase.auth().signInWithPopup(provider).then(result => {
+    const user = result.user;
+    const uid = user.uid;
+    const user_email = user.email;
+    const user_displayName = user.displayName;
+    const user_emailVerified = user.emailVerified;
+    const user_isAnonymous = user.isAnonymous;
+    const user_providerData = user.providerData[0];
+    const user_providerId = user.providerId;
+
+    // check if account already exists, if not add an entry.
+    const accountCheck = firebase.database().ref('users/' + uid);
+    accountCheck.once("value", snapshot => {
+      if (snapshot.val() === null) {
+        accountCheck.set({
+          admin: 'false',
+          email: user_email,
+          emailVerified: user_emailVerified,
+          isAnonymous: user_isAnonymous,
+          providerData: user_providerData,
+          providerID: user_providerId
```

Comment:
it would be better to just assign this fields directly instead of putting them in variables and then assigning those